src/v2/view-builder/internals/BaseModel.js

```diff
@@ -80,6 +87,22 @@ const create = function (remediation = {}, optionUiSchemaConfig = {}) {
   const BaseModel = Model.extend({
     props,
     local,
+    toJSON: function () {
```

Comment:
I'm thinking another option is to call transformUserName at `FormController.handleSaveForm`, right before `idx.proceed`.
This option will work for `processCreds` which could be pure function and async function (return promise)

Reply:
That option is what I originally had - https://github.com/okta/okta-signin-widget/pull/1730/commits/c01f10a98d8939d557aee90045d7fc114caffed1

But I didn't find an existing way to test the FormController, and now I think its easier for testing to put it here and a more intuitive place for it.

processCreds would make sense in the FormController.

Reply:
Hard to test seems not a good justification cause you'll need to test `processCreds` anyway? :D 
In test cafe, We could re-render ui with different set of options so that you'd be able to verify the identifier changed (in request body)

Reply:
> Hard to test seems not a good justification cause you'll need to test `processCreds` anyway? :D
heh I know, but I wanted to put off crossing that bridge until I really had to.